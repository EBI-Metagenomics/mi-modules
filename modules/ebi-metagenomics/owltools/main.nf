
process OWLTOOLS {
    tag "$meta.id"
    label 'process_single'

    container "microbiome-informatics/owltools:2024-06-12"

    input:
    tuple val(meta), path(input_gaf)
    path go_obo
    path goslim_ids

    output:
    tuple val(meta), path("*.gaf"), emit: gaf
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    owltools \\
        ${go_obo} \\
        --gaf ${input_gaf} \\
        $args \\
        --idfile ${goslim_ids} \\
        --write-gaf ${prefix}_goslim_annotations.gaf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        owltools: 2024-06-12
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    touch ${prefix}_goslim_annotations.gaf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        owltools: 2024-06-12
    END_VERSIONS
    """
}
