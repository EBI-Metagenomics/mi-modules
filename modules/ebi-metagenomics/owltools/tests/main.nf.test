
nextflow_process {

    name "Test Process OWLTOOLS"
    script "../main.nf"
    process "OWLTOOLS"
    config "./nextflow.config"

    tag "modules"
    tag "modules_ebimetagenomics"
    tag "owltools"

    test("owltools") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/ERRTESTING_ips_annotations.gaf", checkIfExists: true)
                    ]
                input[1] = 
                    file("${moduleDir}/tests/go-dummy.obo", checkIfExists: true)
                input[2] = 
                    file("${moduleDir}/tests/goslim_ids.txt", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("owltools - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/ERRTESTING_ips_annotations.gaf", checkIfExists: true)
                    ]
                input[1] = 
                    file("${moduleDir}/tests/go-dummy.obo", checkIfExists: true)
                input[2] = 
                    file("${moduleDir}/tests/goslim_ids.txt", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
