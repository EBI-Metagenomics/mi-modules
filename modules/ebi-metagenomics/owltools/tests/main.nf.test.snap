{
    "owltools - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_goslim_annotations.gaf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,15766c0524b190336aac5abe9d9787cc"
                ],
                "gaf": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_goslim_annotations.gaf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,15766c0524b190336aac5abe9d9787cc"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-21T16:40:08.703725323"
    },
    "owltools": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_goslim_annotations.gaf:md5,2c02875f6262e298043cca688907d533"
                    ]
                ],
                "1": [
                    "versions.yml:md5,15766c0524b190336aac5abe9d9787cc"
                ],
                "gaf": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_goslim_annotations.gaf:md5,2c02875f6262e298043cca688907d533"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,15766c0524b190336aac5abe9d9787cc"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-21T16:40:03.117515245"
    }
}